src/borg/fuse.py

```diff
@@ -138,6 +137,23 @@ def _create_dir(self, parent):
         self.parent[ino] = parent
         return ino
 
+    def _create_dir_parents(self, path, parent=1):
+        """Create directory tree (equivalent to mkdir --parents)."""
+        segments = path.split(b'/')
+        segments.reverse()
+        inode = parent
+        while segments:
+            try:
+                inode = self.contents[inode][segments[-1]]
+            except KeyError:
```

Comment:
NB. specifically this is a common pattern in the entire FUSE code via `try: _find_inode except KeyError:`.